baremetal/metal3machine_manager.go

```diff
@@ -1153,22 +1164,22 @@ func (m *MachineManager) clearError() {
 	}
 }
 
-// updateMachineStatus updates a machine object's status.
+// updateMachineStatus updates a Metal3Machine object's status.
 func (m *MachineManager) updateMachineStatus(ctx context.Context, host *bmh.BareMetalHost) error {
 	addrs := m.nodeAddresses(host)
 
-	machineCopy := m.Metal3Machine.DeepCopy()
-	machineCopy.Status.Addresses = addrs
+	metal3MachineOld := m.Metal3Machine.DeepCopy()
 
-	if equality.Semantic.DeepEqual(m.Machine.Status, machineCopy.Status) {
+	m.Metal3Machine.Status.Addresses = addrs
+	conditions.MarkTrue(m.Metal3Machine, capm3.AssociateBMHCondition)
```

Comment:
does it mean that we are telling a  bmh is now associated if metal3machine `status.addresses` is populated?

Reply:
Yes. It's called by Update() which updates the BMH object and then calls this func to update the Meal3Machine obj.